### ISSUE TITLE
Gradle 8.8 causing Java crashes on older macOS

### ISSUE THREAD

**Issue by p3k** [gradle/gradle#29555]
### Current Behavior

I am seeing Java crashes when building my project with Gradle:

<details><summary>Click to expand</summary>

```shell
$ ./gradlew build
Starting a Gradle Daemon, 10 busy Daemons could not be reused, use --status for details
> Task :hopKit:compileJava SKIPPED
> Task :jala:compileJava SKIPPED
> Task :hopKit:processResources SKIPPED
> Task :jala:processResources SKIPPED
> Task :hopKit:classes UP-TO-DATE
> Task :jala:classes UP-TO-DATE
> Task :hopKit:jar SKIPPED
> Task :jala:jar SKIPPED
> Task :hopKit:assemble UP-TO-DATE
> Task :jala:assemble UP-TO-DATE
> Task :jala:compileTestJava SKIPPED
> Task :hopKit:compileTestJava SKIPPED
> Task :jala:processTestResources SKIPPED
> Task :hopKit:processTestResources SKIPPED
> Task :jala:testClasses UP-TO-DATE
> Task :hopKit:testClasses UP-TO-DATE
> Task :jala:test SKIPPED
> Task :hopKit:test SKIPPED
> Task :jala:check UP-TO-DATE
> Task :hopKit:check UP-TO-DATE
> Task :jala:build UP-TO-DATE
> Task :hopKit:build UP-TO-DATE
> Task :modules:compileJava SKIPPED
> Task :test:compileJava SKIPPED
> Task :modules:processResources SKIPPED
> Task :test:processResources SKIPPED
> Task :modules:classes UP-TO-DATE
> Task :test:classes UP-TO-DATE
> Task :modules:jar SKIPPED
> Task :test:jar SKIPPED
> Task :modules:assemble UP-TO-DATE
> Task :test:assemble UP-TO-DATE
> Task :test:compileTestJava SKIPPED
> Task :modules:compileTestJava SKIPPED
> Task :modules:processTestResources SKIPPED
> Task :test:processTestResources SKIPPED
> Task :test:testClasses UP-TO-DATE
> Task :modules:testClasses UP-TO-DATE
> Task :test:test SKIPPED
> Task :test:check UP-TO-DATE
> Task :test:build UP-TO-DATE
> Task :modules:test SKIPPED
> Task :modules:check UP-TO-DATE
> Task :modules:build UP-TO-DATE
The message received from the daemon indicates that the daemon has disappeared.
Build request sent: Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma}
Attempting to read last messages from the daemon log...
Daemon pid: 99864
  log file: /Users/tobi/.config/gradle/daemon/8.8/daemon-99864.out.log
----- Last  20 lines from daemon log file - daemon-99864.out.log -----
2024-06-14T22:57:00.076+0200 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 18: received class org.gradle.launcher.daemon.protocol.Build
2024-06-14T22:57:00.077+0200 [DEBUG] [org.gradle.launcher.daemon.server.DefaultDaemonConnection] thread 18: Received non-IO message from client: Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma}
2024-06-14T22:57:00.078+0200 [INFO] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Received command: Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma}.
2024-06-14T22:57:00.078+0200 [DEBUG] [org.gradle.launcher.daemon.server.DefaultIncomingConnectionHandler] Starting executing command: Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma} with connection: socket connection from /127.0.0.1:53442 to /127.0.0.1:53443.
2024-06-14T22:57:00.082+0200 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] Command execution: started DaemonCommandExecution[command = Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma}, connection = DefaultDaemonConnection: socket connection from /127.0.0.1:53442 to /127.0.0.1:53443] after 0.0 minutes of idle
2024-06-14T22:57:00.082+0200 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Marking the daemon as busy, address: [c0cd8ec9-1c64-4873-bff9-8515e72f7819 port:53442, addresses:[localhost/127.0.0.1]]
2024-06-14T22:57:00.082+0200 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Marking busy by address: [c0cd8ec9-1c64-4873-bff9-8515e72f7819 port:53442, addresses:[localhost/127.0.0.1]]
2024-06-14T22:57:00.083+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on daemon addresses registry.
2024-06-14T22:57:00.083+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
2024-06-14T22:57:00.086+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
2024-06-14T22:57:00.086+0200 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
2024-06-14T22:57:00.087+0200 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
2024-06-14T22:57:00.089+0200 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=154e88c4-3bee-4477-b9ad-698cf0edad87, currentDir=/Users/tobi/Projects/helma}. Dispatching build started information...
2024-06-14T22:57:00.090+0200 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 20: dispatching org.gradle.launcher.daemon.protocol.BuildStarted@7fbb857c
2024-06-14T22:57:00.094+0200 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: [PATH, __impureHostDeps, GETTEXTDATADIRS, FNM_COREPACK_ENABLED, depsBuildTarget, buildInputs, doCheck, DEVBOX_CONFIG_DIR, __propagatedSandboxProfile, NIX_PROFILES, DEVBOX_PACKAGES_DIR, FNM_NODE_DIST_MIRROR, SSH_AGENT_PID, COLORTERM, SDKMAN_DIR, depsHostHostPropagated, PWD, nativeBuildInputs, TERM_PROGRAM_VERSION, shellHook, FNM_LOGLEVEL, stdenv, patches, SIZE, SDKMAN_CANDIDATES_API, FNM_VERSION_FILE_STRATEGY, dontAddDisableDepTrack, IN_NIX_SHELL, AR, AS, XPC_FLAGS, system, DETERMINISTIC_BUILD, LC_CTYPE, NIX_SSL_CERT_FILE, SHLVL, FNM_RESOLVE_ENGINES, STRIP, NIX_ENFORCE_NO_NATIVE, MallocNanoZone, HOST_PATH, JAVA_HOME, cmakeFlags, TERM, PIP_REQUIRE_VIRTUALENV, depsTargetTarget, propagatedBuildInputs, out, JAVA_MAIN_CLASS_99852, VENV_DIR, PKG_CONFIG_PATH_FOR_TARGET, builder, SDKMAN_OFFLINE_MODE, phases, __sandboxProfile, CC, CADDY_ROOT_DIR, PYTHONNOUSERSITE, FNM_MULTISHELL_PATH, _tide_color_separator_same_color, TERM_PROGRAM, DEVBOX_PATH_STACK, DEVBOX_SYSTEM_SED, ZERO_AR_DATE, __DEVBOX_SHELLENV_HASH_da7a34312cc66be048ba49740b70f2938984582c957d4494be76b9b86706a809, DEVBOX_REFRESH_ALIAS_da7a34312cc66be048ba49740b70f2938984582c957d4494be76b9b86706a809, CADDY_LOG_DIR, propagatedNativeBuildInputs, LAUNCHER_VERSION, LD, SDKMAN_ENV, NIX_HARDENING_ENABLE, NIX_BINTOOLS_WRAPPER_TARGET_HOST_x86_64_apple_darwin, outputs, CXX, DEVBOX_NIX_ENV_PATH_da7a34312cc66be048ba49740b70f2938984582c957d4494be76b9b86706a809, XDG_DATA_DIRS, NIX_IGNORE_LD_THROUGH_GCC, strictDeps, CADDY_CONFIG, LAUNCHER_PATH, _tide_location_color, LOGNAME, depsTargetTargetPropagated, PYTHONPATH, RANLIB, SHELL, __ETC_PROFILE_NIX_SOURCED, LD_DYLD_PATH, FNM_ARCH, doInstallCheck, NIX_CC, DEVBOX_SYSTEM_BASH, mesonFlags, TMPDIR, buildPhase, ORIGINAL_XDG_CURRENT_DESKTOP, GNUPGHOME, shell, NIX_BINTOOLS, __CF_USER_TEXT_ENCODING, name, NIX_LDFLAGS, __structuredAttrs, depsBuildTargetPropagated, depsHostHost, MACOSX_DEPLOYMENT_TARGET, NM, STRINGS, NIX_NO_SELF_RPATH, FNM_DIR, DEVBOX_PROJECT_ROOT, configureFlags, PATH_LOCALE, LANG, DISPLAY, DEVBOX_INIT_PATH, PKG_CONFIG_FOR_TARGET, CONFIG_SHELL, SOURCE_DATE_EPOCH, XPC_SERVICE_NAME, PYTHONHASHSEED, SDKMAN_CANDIDATES_DIR, __propagatedImpureHostDeps, NIX_DONT_SET_RPATH_FOR_BUILD, NIX_DONT_SET_RPATH, __darwinAllowLocalNetworking, NIX_CFLAGS_COMPILE, GPG_AGENT_INFO, preferLocalBuild, GPG_TTY, SDKMAN_PLATFORM, depsBuildBuildPropagated, USER, NIX_PKG_CONFIG_WRAPPER_TARGET_TARGET_x86_64_apple_darwin, SSH_AUTH_SOCK, EDITOR, NIX_STORE, NIX_CC_WRAPPER_TARGET_HOST_x86_64_apple_darwin, depsBuildBuild, HOME, NIX_BUILD_CORES]
2024-06-14T22:57:00.109+0200 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
2024-06-14T22:57:00.109+0200 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 99864). The daemon log file: /Users/tobi/.config/gradle/daemon/8.8/daemon-99864.out.log
2024-06-14T22:57:00.119+0200 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting build in new daemon [memory: 2 GiB]
2024-06-14T22:57:00.122+0200 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
2024-06-14T22:57:00.122+0200 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=39e68a53-6486-48d7-8d3b-4557d8d12aa6,javaHome=/Users/tobi/.sdkman/candidates/java/11.0.23-tem,javaVersion=11,daemonRegistryDir=/Users/tobi/.config/gradle/daemon,pid=99864,idleTimeout=10800000,priority=NORMAL,applyInstrumentationAgent=true,nativeServicesMode=ENABLED,daemonOpts=-XX:+HeapDumpOnOutOfMemoryError,--add-opens=java.base/java.util=ALL-UNNAMED,--add-opens=java.base/java.lang=ALL-UNNAMED,--add-opens=java.base/java.lang.invoke=ALL-UNNAMED,--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens=java.base/java.nio.charset=ALL-UNNAMED,--add-opens=java.base/java.net=ALL-UNNAMED,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,-Xmx2g,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant]
----- End of the daemon log -----

JVM crash log found: file:///Users/tobi/Projects/helma/hs_err_pid99864.log

FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
> Task :launcher:compileJava
> Task :generateLicenseReport
```
</details>

### Expected Behavior

This the output of the same command with Gradle 8.7:

<details><summary>Click to expand</summary>

```shell
$ ./gradlew build
> Task :jala:compileJava SKIPPED
> Task :hopKit:compileJava SKIPPED
> Task :hopKit:processResources SKIPPED
> Task :jala:processResources SKIPPED
> Task :hopKit:classes UP-TO-DATE
> Task :jala:classes UP-TO-DATE
> Task :hopKit:jar SKIPPED
> Task :jala:jar SKIPPED
> Task :hopKit:assemble UP-TO-DATE
> Task :jala:assemble UP-TO-DATE
> Task :jala:compileTestJava SKIPPED
> Task :jala:processTestResources SKIPPED
> Task :hopKit:compileTestJava SKIPPED
> Task :jala:testClasses UP-TO-DATE
> Task :hopKit:processTestResources SKIPPED
> Task :hopKit:testClasses UP-TO-DATE
> Task :jala:test SKIPPED
> Task :jala:check UP-TO-DATE
> Task :jala:build UP-TO-DATE
> Task :hopKit:test SKIPPED
> Task :hopKit:check UP-TO-DATE
> Task :modules:compileJava SKIPPED
> Task :hopKit:build UP-TO-DATE
> Task :modules:processResources SKIPPED
> Task :modules:classes UP-TO-DATE
> Task :test:compileJava SKIPPED
> Task :modules:jar SKIPPED
> Task :modules:assemble UP-TO-DATE
> Task :test:processResources SKIPPED
> Task :test:classes UP-TO-DATE
> Task :modules:compileTestJava SKIPPED
> Task :test:jar SKIPPED
> Task :modules:processTestResources SKIPPED
> Task :test:assemble UP-TO-DATE
> Task :modules:testClasses UP-TO-DATE
> Task :test:compileTestJava SKIPPED
> Task :test:processTestResources SKIPPED
> Task :test:testClasses UP-TO-DATE
> Task :modules:test SKIPPED
> Task :test:test SKIPPED
> Task :test:check UP-TO-DATE
> Task :modules:check UP-TO-DATE
> Task :test:build UP-TO-DATE
> Task :modules:build UP-TO-DATE
> Task :launcher:compileJava FROM-CACHE
> Task :launcher:processResources NO-SOURCE
> Task :launcher:classes UP-TO-DATE
> Task :launcher:jar
> Task :launcher:assemble
> Task :launcher:compileTestJava NO-SOURCE
> Task :launcher:processTestResources NO-SOURCE
> Task :launcher:testClasses UP-TO-DATE
> Task :launcher:test NO-SOURCE
> Task :launcher:check UP-TO-DATE
> Task :launcher:build
> Task :generateLicenseReport FROM-CACHE
> Task :processSource
> Task :compileJava FROM-CACHE
> Task :processResources
> Task :classes
> Task :jar
> Task :javadoc FROM-CACHE
> Task :jsdoc
> Task :startScripts
> Task :distTar
> Task :distZip
> Task :assemble
> Task :compileTestJava NO-SOURCE
> Task :processTestResources NO-SOURCE
> Task :testClasses UP-TO-DATE
> Task :test NO-SOURCE
> Task :check UP-TO-DATE
> Task :build

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 26s
12 actionable tasks: 8 executed, 4 from cache
```
</details>

### Context (optional)

I am aware this is an issue with an older macOS on an even older machine. Nevertheless, the issue surfaced between two minor versions – thus, it might point to a general issue on the platform.

### Steps to Reproduce

1. Checkout https://github.com/antville/helma
   ```shell
   git clone https://github.com/antville/helma.git
   cd helma
   ```
2. Run the Gradle build task
   ```shell
   ./gradlew build
   ```

### Gradle version

8.8

### Gradle version that used to work

8.7

### Your Environment (optional)

```shell
$ system_profiler SPSoftwareDataType SPHardwareDataType -detailLevel mini
Software:

    System Software Overview:

      System Version: macOS 10.15.7 (19H2026)
      Kernel Version: Darwin 19.6.0
      Time since boot: 8:37

Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro9,2
      Processor Name: Dual-Core Intel Core i7
      Processor Speed: 2,9 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 4 MB
      Hyper-Threading Technology: Enabled
      Memory: 16 GB
      Boot ROM Version: 429.0.0.0.0
      SMC Version (system): 2.2f44
      Sudden Motion Sensor:
          State: Enabled
```

**Comment by ljacomet** [comment]
This issue will be closed as a duplicate of

* #29476 

If you think our analysis is wrong, please provide us with more detailed information explaining why.

***

See the comments there for workarounds.

**Comment by p3k** [comment]
A fix was merged, but not sure if this patches 8.8: https://github.com/gradle/gradle/pull/29514
